mdsuite/calculators/coordination_number_calculation.py

```diff
@@ -109,267 +140,279 @@ def __init__(self, **kwargs):
+        coordination_numbers : dict
+                A dictionary of coordination numbers.
         """
 
-        self.indices = self._find_minima()  # get the minimums
+        coordination_shells = self._find_minima(radii, rdf)  # get the minimums
 
-        # Calculate the coordination numbers by averaging over the two values
-        # returned by _find_minimums
-        first_shell = np.mean(
-            [
-                self.integral_data[self.indices[0][0]],
-                self.integral_data[self.indices[0][1]],
-            ]
-        )
-        first_shell_error = (
-            np.std(
-                [
-                    self.integral_data[self.indices[0][0]],
-                    self.integral_data[self.indices[0][1]],
-                ]
+        coordination_numbers = {}
+
+        for key, val in coordination_shells.items():
+            lower_bound = integral_data[val[0]]
+            upper_bound = integral_data[val[1]]
+
+            coordination_numbers[f"CN_{int(key) + 1}"] = np.mean(
+                [lower_bound, upper_bound]
             )
-            / np.sqrt(2)
-        )
+            coordination_numbers[f"CN_{int(key) + 1}_error"] = np.std(
+                [lower_bound, upper_bound]
+            ) / np.sqrt(2)
 
-        # # TODO what about second shell?
-        # second_shell = (
-        #     np.mean(
-        #         [
-        #             self.integral_data[self.indices[1][0]],
-        #             self.integral_data[self.indices[1][1]],
-        #         ]
-        #     )
-        #     - first_shell
-        # )
-        # second_shell_error = (
-        #     np.std(
-        #         [
-        #             self.integral_data[self.indices[1][0]],
-        #             self.integral_data[self.indices[1][1]],
-        #         ]
-        #     )
-        #     / np.sqrt(2)
-        # )
-
-        return first_shell, first_shell_error
+        return coordination_numbers
 
     def run_calculator(self):
         """
         Calculate the coordination numbers and perform error analysis
         """
-        calculations = self.experiment.run.RadialDistributionFunction(plot=False)
-        self.data_range = calculations.data_range
-        for (
-            selected_species,
-            vals,
-        ) in calculations.data_dict.items():  # Loop over all existing RDFs
+
+        for selected_species, vals in self.rdf_data.data_dict.items():
             log.debug(f"Computing coordination numbers for {selected_species}")
-            self.radii = np.array(vals["x"]).astype(float)[1:]
-            self.rdf = np.array(vals["y"]).astype(float)[1:]
-            self.selected_species = selected_species.split("_")
-            self.species_tuple = selected_species  # depreciated
 
-            density = self._get_density(self.selected_species[0])  # calculate the density
+            radii = np.array(vals["x"]).astype(float)[1:]
+            rdf = np.array(vals["y"]).astype(float)[1:]
```

Comment:
This should use `RadialDistributionFunction.series_keys` so that we don't have to rely on it being always `x` and `y`.
Maybe it would be best if we added the keys as class attributes so that they can be read from the class without init?